### ISSUE TITLE
Format of recarray when reading IDL structures with readsav()

### ISSUE THREAD

**Issue by rmchurch** [scipy/scipy#5480]
When saving an IDL structure to a .sav file, then reading with scipy.io.idl.readsav(), the resulting Numpy recarray has an additional array index that is annoying. For example:

IDL:
IDL> s = {x:[0,1,2,3], y:[5,6,7,8]}
IDL> print,s.x
       0       1       2       3
IDL>print, s.x[0]
       0

But in Python, if I read in this IDL structure saved in a .sav file:
In [3]: from scipy.io.idl import readsav
In [4]: recs = readsav('file.sav')
In [5]: rec.s.x
Out[5]: array([array([0, 1, 2, 3], dtype=int16)], dtype=object)
In [6]: rec.s.x[0]
Out[6]: array([0, 1, 2, 3], dtype=int16)
In [7]: rec.s.x[0][0]
Out[7]: 0

There is an extra array on top of the 'x' variable in the s struct, so to access elements I have to add an extra [0] index. I'm not sure why this dtype=object array is necessary, but I would much prefer a similar the structs come into Python with a similar indexing experience to IDL.

**Comment by rgommers** [comment]
Hmm, no idea why. But probably tricky to change without breaking existing code. @astrofrog?

**Comment by astrofrog** [comment]
Yes, while I agree this would be nice, it would break backward-compatibility for a lot of people since this is a common use-case. We could consider adding an option to ``readsav`` though.

**Comment by rmchurch** [comment]
A keyword option would be fine, I would hate to break backwards compatibility.
